torch/csrc/api/include/torch/nn/utils/clip_grad.h

```diff
@@ -14,7 +14,12 @@ inline double clip_grad_norm_(
     std::vector<Tensor> parameters,
     double max_norm,
     double norm_type = 2.0,
-    bool error_if_nonfinite = true) {
+    bool error_if_nonfinite = false) {
+  if (!error_if_nonfinite) {
```

Comment:
I feel like we only want to add this warning if it actually goes to infinite? Otherwise we might spam users for no reason as they passed no argument and they didn't had any pathological value.

Reply:
Cool, I think that's a good idea. Only thing I'll mention is that currently the check isn't being done unless the flag is set, so hopefully there's no big performance hit from the extra call (there shouldn't be as far as I can tell).